measurements/api/measurements.py

```diff
@@ -361,26 +423,83 @@ def list_measurements(
     if order_by is not None:
         q = q.order_by(text('{} {}'.format(order_by, order)))
 
-    query_time = 0
+
+    ## Create a query for fastpath
+
+    fpcols = [
+        func.coalesce(0).label('m_input_no'),
+        Fastpath.measurement_start_time.label("measurement_start_time"),
+        func.concat("tid-", Fastpath.tid).label('measurement_id'),
+        func.coalesce(0).label('m_report_no'),
+
+        func.coalesce(false()).label('anomaly'),
+        func.coalesce(false()).label('confirmed'),
+        func.coalesce(false()).label('msm_failure'),
+        cast(Fastpath.scores.label("scores"), String),
+
+        func.coalesce([0, ]).label('exc'),
+        func.coalesce(0).label('residual_no'),
+
+        Fastpath.report_id.label("report_id"),
+        Fastpath.probe_cc.label("probe_cc"),
+        Fastpath.probe_asn.label("probe_asn"),
+        Fastpath.test_name.label("test_name"),
+        func.coalesce(0).label('report_no'),
+    ]
+    if cte is None:
+        fpcols.append(Fastpath.input.label("input"))
+        assert len(fpcols) == len(cols)
+    else:
+        fpcols.append(Fastpath.input.label("input_cte_input"))
+        fpcols.append(func.coalesce(0).label('input_cte_input_no'))
+        assert len(fpcols) == len(cols) + 1
+
+    fpq = current_app.db_session.query(*fpcols)
+
+    if since is not None:
+        fpq = fpq.filter(Fastpath.measurement_start_time > since)
+    if until is not None:
+        fpq = fpq.filter(Fastpath.measurement_start_time <= until)
+    if report_id:
+        fpq = fpq.filter(Fastpath.report_id == report_id)
+    if probe_cc:
+        fpq = fpq.filter(Fastpath.probe_cc == probe_cc)
+    if probe_asn is not None:
+        fpq = fpq.filter(Fastpath.probe_asn == probe_asn)
+    if test_name is not None:
+        fpq = fpq.filter(Fastpath.test_name == test_name)
+    if input_:
+        fpq = fpq.filter(Fastpath.input == input_)
+
+    # Assemble the query
+
     q = q.limit(limit).offset(offset)
+    try:
+        q = q.union(fpq)
```

Comment:
How do you ensure that a record is not duplicated inside of the fastpath and the batch pipeline and hence you will return duplicate rows by doing a UNION?

Reply:
Updated to return only one (merged) result